src/audio/copier/copier.c

```diff
@@ -397,11 +397,11 @@ static int create_dai(struct comp_dev *parent_dev, struct copier_data *cd,
 		ret = init_dai(parent_dev, drv, config, copier, pipeline, &dai, type, i);
 		if (ret) {
 			comp_err(parent_dev, "failed to create dai");
-			break;
+			return ret;
```

Comment:
the commit description isn't quite consistent with the code - `dai_count` cannot be 0 here. It's 1 initially and can only change in the ALH case at line 372 above, but there you already check for 0.